### PR TITLE
Define a basic compliance pipeline

### PR DESCRIPTION
**Pull request by tmeschter**
Define a basic pipeline that runs PoliCheck.

The point here is just to get PoliCheck to run. Later commits will expand on this.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7577)

### CODE REVIEW

File: build/ci/compliance.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+#
+# NOTE: triggers for this build are defined in the Web UI instead of here in the YAML file so they
+#       apply to all branches.
+
+resources:
+- repo: self
+  clean: true
+queue:
+  name: VSEngSS-MicroBuild2019
```
Comment (swesonga) [inline build/ci/compliance.yml:14]: Should this use the new 1ES pool?

**Review by swesonga — APPROVED** [review]

File: build/ci/compliance.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+
+steps:
+- script: echo Hello, world!
+  displayName: 'Run a one-line script'
+
+- script: |
+    echo Add other tasks to build, test, and deploy your project.
+    echo See https://aka.ms/yaml
+  displayName: 'Run a multi-line script'
```
Comment (MiYanni) [inline build/ci/compliance.yml:30]: Looks like generic template text that doesn't need to be here.

**Review by MiYanni — COMMENTED** [review]
I fully expect that we'll also need exception files (those that contain things to skip) for these checks. Every repo generally has a few things that need to be ignored.

File: build/ci/compliance.yml (hunk 1 above)
Comment (tmeschter) [reply build/ci/compliance.yml:14]: Yeah, it will eventually.

File: build/ci/compliance.yml (hunk 2 above)
Comment (tmeschter) [reply build/ci/compliance.yml:30]: Yeah, I left those there at first just to make sure something was running.

File: build/ci/compliance.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+- task: APIScan@2
+  inputs:
+    softwareFolder: '$(Build.SourcesDirectory)\artifacts'
+    softwareName: 'Dotnet-Project-System'
+    softwareVersionNum: '17.0'
+    softwareBuildNum: '$(Build.BuildId)'
+    symbolsFolder: 'SRV*http://symweb'
+
+- task: TSAUpload@2
```
Comment (MiYanni) [inline build/ci/compliance.yml:47]: What is TSAUpload? I'm having difficulty finding docs on it.

Comment (tmeschter) [reply build/ci/compliance.yml:47]: It picks up the output files from the previous stages and uses them to auto-generate bugs.

**Review by jjmew — CHANGES_REQUESTED** [review]
:clock1:

**Review by jjmew — APPROVED** [review]
:shipit:

**Pull request merged**